### ISSUE TITLE
Flutter build for iOS failing after removing a dependency package entry from Pubspec file

### ISSUE THREAD

**Issue by Roshanalexander** [flutter/flutter#61063]
After removing a dependency package 'fluttertoast' from the pubspec file and updated few other ones, the iOS build stopped working. Android build still works fine. I've attempted some steps which are listed below but still not working.

## Steps I've attempted:
```
flutter clean
flutter pub cache repair
flutter pub get
flutter run --debug
```

## Error Output
```
Launching lib/main.dart on iPhone 11 Pro in debug mode...
Running pod install...                                              3.3s
Running Xcode build...                                                  
                                                   
 └─Compiling, linking and signing...                         2.5s
Xcode build done.                                           21.7s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    /Users/ra/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.15.1/darwin/Classes/AudioplayersPlugin.m:88:37: warning: incompatible pointer types sending 'FlutterEngine *' to parameter of type 'NSObject<FlutterBinaryMessenger> *
    _Nonnull' [-Wincompatible-pointer-types]
                        binaryMessenger:_headlessEngine];
                                        ^~~~~~~~~~~~~~~
    In module 'Flutter' imported from /Users/ra/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.15.1/darwin/Classes/AudioplayersPlugin.h:2:
    /Users/ra/Documents/Workspace/Flutter Projects/test/ios/Flutter/Flutter.framework/Headers/FlutterChannels.h:178:74: note: passing argument to parameter 'messenger' here
                          binaryMessenger:(NSObject<FlutterBinaryMessenger>*)messenger;
                                                                             ^
    1 warning generated.
    /Users/ra/Documents/Workspace/Flutter Projects/test/ios/Runner/AppDelegate.swift:13:58: warning: conditional cast from 'AppDelegate' to 'UNUserNotificationCenterDelegate' always succeeds
          UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
                                                             ^
    ld: framework not found fluttertoast
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

Could not build the application for the simulator.
Error launching application on iPhone 11 Pro.

```

## Flutter Doctor
```
[✓] Flutter (Channel stable, v1.17.5, on Mac OS X 10.15.5 19F101, locale en-US)
    • Flutter version 1.17.5 at /Users/ra/Developer/flutter
    • Framework revision 8af6b2f038 (7 days ago), 2020-06-30 12:53:55 -0700
    • Engine revision ee76268252
    • Dart version 2.8.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/ra/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.3

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[!] IntelliJ IDEA Ultimate Edition (version 2020.1.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.dev/intellij-setup/#installing-the-plugins

[✓] VS Code (version 1.46.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.12.1

[✓] Connected device (1 available)
    • iPhone 11 Pro • 4871732D-0CB6-494D-A214-14F3AA4132FF • ios • com.apple.CoreSimulator.SimRuntime.iOS-13-5 (simulator)
```

Also, I ran the build directly inside Xcode and it says `"Framework not found fluttertoast"`. This is the package I just removed from pubspec. Somehow it seems something is left over.

![Screen Shot 2020-07-07 at 10 47 52 PM](https://user-images.githubusercontent.com/6432696/86869773-27515580-c0a5-11ea-9127-22688a1757ec.png)

**Comment by TahaTesser** [comment]
Hi @Roshanalexander 
Whenever you remove a package, it's good to idea to update your pod files, please `execute `pod update` from root of your `ios` flutter
if the problem problem
Delete podfiles by executing  `rm -rf podfile` and then `pod init` follow up with `pod install`

This is just warning for your native code and  `AppDelegate.swift` class, you can easily fix last warning `self as? UNUserNotificationCenterDelegate`, open your ios project in xocde and go to `AppDelegate`

```

    /Users/ra/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.15.1/darwin/Classes/AudioplayersPlugin.m:88:37: warning: incompatible pointer types sending 'FlutterEngine *' to parameter of type 'NSObject<FlutterBinaryMessenger> *
    _Nonnull' [-Wincompatible-pointer-types]
                        binaryMessenger:_headlessEngine];
                                        ^~~~~~~~~~~~~~~
    In module 'Flutter' imported from /Users/ra/Developer/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.15.1/darwin/Classes/AudioplayersPlugin.h:2:
    /Users/ra/Documents/Workspace/Flutter Projects/test/ios/Flutter/Flutter.framework/Headers/FlutterChannels.h:178:74: note: passing argument to parameter 'messenger' here
                          binaryMessenger:(NSObject<FlutterBinaryMessenger>*)messenger;
                                                                             ^
    1 warning generated.
    /Users/ra/Documents/Workspace/Flutter Projects/test/ios/Runner/AppDelegate.swift:13:58: warning: conditional cast from 'AppDelegate' to 'UNUserNotificationCenterDelegate' always succeeds
          UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
   1 warning generated.
    /Users/ra/Documents/Workspace/Flutter Projects/test/ios/Runner/AppDelegate.swift:13:58: warning: conditional cast from 'AppDelegate' to 'UNUserNotificationCenterDelegate' always succeeds
          UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
                                                             ^
```

If the problem persist after these tried, your `flutter run --verbose` and a complete reproducible minimal code sample
Thank you

**Comment by Roshanalexander** [comment]
Hi @TahaTesser,

I followed your steps but still getting the error.

```
Launching lib/main.dart on iPhone 11 Pro in debug mode...
Running pod install...                                              1.2s
Running Xcode build...
Xcode build done.                                           18.3s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    /Users/roshanalexander/Documents/Workspace/Flutter Projects/Segtho/ios/Runner/GeneratedPluginRegistrant.m:10:9: fatal error: module 'app_settings' not found
    @import app_settings;
     ~~~~~~~^~~~~~~~~~~~
    1 error generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

Could not build the application for the simulator.
Error launching application on iPhone 11 Pro.
```

**Comment by Roshanalexander** [comment]
[flutter run --verbose output.txt](https://github.com/flutter/flutter/files/4891384/flutter.run.--verbose.output.txt)

**Comment by Roshanalexander** [comment]
@TahaTesser I just put back the dependency package that I removed from pubspec.yaml and it started building fine. Then I removed the package again and ran the build, it failed. This looks to me like, when a package is removed from the pubspec file, the files specific to iOS aren't getting updated. Xcode still thinks that package is included but it wouldn't be able to find it.

**Comment by jmagman** [comment]
@Roshanalexander Can we get your pubspec.yaml?

**Comment by Roshanalexander** [comment]
Here it is. I renamed it to Pubspec.txt to be able to upload here.

[pubspec.txt](https://github.com/flutter/flutter/files/4922244/pubspec.txt)

**Comment by Roshanalexander** [comment]
Another observation: Even updating the version number for fluttertoast package seems to break my iOS build.

**Comment by markusaksli-nc** [comment]
@Roshanalexander I was able to reproduce this issue by adding and removing one dependency from a fresh project. After running `pub get`, `pub upgrade`, and then simply `pod install` in the ios folder instead of removing the podfile first the issue was fixed. Could you try this approach as a temporary workaround and let me know if it works?

Marking as reproducible on latest master `1.21.0-8.0.pre.122`.

<details>
 <summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 1.21.0-8.0.pre.122, on Mac OS X 10.15.6 19G73, locale en-GB)
    • Flutter version 1.21.0-8.0.pre.122 at /Users/markus/development/flutter_master
    • Framework revision 7c6f9dd239 (19 hours ago), 2020-08-09 10:31:03 -0400
    • Engine revision 6d86e67f04
    • Dart version 2.10.0 (build 2.10.0-4.0.dev 0341576448)


[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at /Users/markus/Library/Android/sdk
    • Platform android-30, build-tools 30.0.1
    • Java binary at: /Users/markus/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/193.6626763/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Users/markus/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/193.6626763/Android Studio.app/Contents
    • Flutter plugin version 48.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)


[✓] Connected device (5 available)
    • sdk gphone x86 arm (mobile) • emulator-5554                            • android-x86    • Android 11 (API 30) (emulator)
    • Nevercode’s iPhone (mobile) • b668e524315069f3db3661ac11ff1f66afafebdb • ios            • iOS 13.6
    • macOS (desktop)             • macos                                    • darwin-x64     • Mac OS X 10.15.6 19G73
    • Web Server (web)            • web-server                               • web-javascript • Flutter Tools
    • Chrome (web)                • chrome                                   • web-javascript • Google Chrome 84.0.4147.105

• No issues found!
```
</details>

**Comment by jersocOD** [comment]
Hi guys, after a couple of days trying to figure out how to solve this problem, finally found a solution:
### Solution:

1. Go to Xcode
2. Select your **target**
3. Go to **Build Settings**
4. **Search** the package name shown in the error. e. g. : framework not found **fluttertoast**
5. You will see that there is a section like: "**Linking**"->"**Other linker flags**" ,then **click the bold part** of this row, It will open a dialog box where you can **remove the framework not found**.
6. When removing the framework name, also **remove the "framework" tag**

Reference:https://stackoverflow.com/questions/36904636/ld-framework-not-found-framework-error

**Comment by pranavchunduru3** [comment]
I am running into a similar problem. Did you find a solution?

**Comment by jmagman** [comment]
@jersocOD There shouldn't be any explicit linking to plugins in the Flutter project.  It's managed by CocoaPods, so any changes you make to the `Pods` target will be regenerated after a `flutter clean`.

**Comment by jmagman** [comment]
I'm not sure what the original `fluttertoast` issue was, there's no verbose logs from that happening so I don't think this issue is actionable.  



The subsequent `app_settings` is likely because that's a Swift plugin, but you have an Objective-C app.  In which case you need to add `use_frameworks!` to your `ios/Podfile` as the plugin author noted in their README https://github.com/spencerccf/app_settings#ios.

If anyone is seeing linker issues after removing a plugin from your project, please file a new issue and completely fill out the template.  Also include a `flutter run` logs with the `-v` verbose flag (it will be large, so you'll need to attach it).

Closing.

**Comment by Roshanalexander** [comment]
@jmagman This issue was not completely on FlutterToast package. This issue happens randomly when you try to remove a package from the yaml file and try to run the app on iOS. Not sure how you found that I'm using an objective C app. I have not done anything specific on iOS side to create the app. Just generate the project as a flutter project. Seems like when you remove a package from the yaml file, the libs on the iOS side are not cleaned up properly which causes iOS build to fail. Android works fine.

**Comment by Roshanalexander** [comment]
> I am running into a similar problem. Did you find a solution?

Hey, sorry for the late reply. This was a random error which happened to me. So I left the packages untouched to prevent the issue from occurring even though I don't use them.

**Comment by jmagman** [comment]
> @jmagman This issue was not completely on FlutterToast package. This issue happens randomly when you try to remove a package from the yaml file and try to run the app on iOS. 

@Roshanalexander Sorry, I was too hasty closing this, I should have asked if you were still seeing this issue since a lot of the linker/Flutter plugin linking code has been rewritten since you filed this.

> Not sure how you found that I'm using an objective C app. I have not done anything specific on iOS side to create the app. Just created the project as a regular flutter project which I believe was using Swift.

I see `SWIFT_VERSION = 5.0` in your logs now that I'm looking for it. 🙂 

> To me, it Seems like when you remove a package from the yaml file, the libs on the iOS side are not cleaned up properly which causes iOS build to fail. Android works fine.

> Btw, use_frameworks! is already defined in my pod file.

👍 

Can you attach your `ios/Podfile` and an updated `flutter doctor -v`?  And next time you see this (or if you have time to reproduce this), can you attach:

1. `.flutter-plugins-dependencies`
2. `ls -la  ios/.symlinks/plugins/`

**Comment by Lourie-120** [comment]
<img width="1380" alt="스크린샷 2021-04-21 오전 11 40 34" src="https://user-images.githubusercontent.com/82920988/115489018-6f3d1400-a296-11eb-8d51-dba1c43d9688.png">

I deleted it while using "flt_telephony_info", and I had the same problem.
I searched for the package name in xcode and found it in Runner.xcodeproj - Target Runner - Build Setting.
Linking - Other Linker Flags cleared the package and successfully built.

I don't know if this is the right solution, but I hope it helps you.

**Comment by jmagman** [comment]
@Lourie-120 Can you check your source control to find when those `OTHER_LDFLAGS ` linker flags were added?  Flutter wouldn't have added them, CocoaPods handles linking with plugins, and it doesn't edit the Runner target at all to do it.

If you added them yourself I'm curious why, and if you did, you would be responsible for maintaining those flags...

**Comment by Roshanalexander** [comment]
> Can you attach your `ios/Podfile` and an updated `flutter doctor -v`? And next time you see this (or if you have time to reproduce this), can you attach:
> 
> 1. `.flutter-plugins-dependencies`
> 2. `ls -la  ios/.symlinks/plugins/`

Hi @jmagman, here's the requested items:

**flutter doctor -v Output:**
```
roshanalexander@Roshans-MBP Project % flutter doctor -v      
[✓] Flutter (Channel stable, 2.0.5, on macOS 11.2.3 20D91 darwin-x64, locale en-US)
    • Flutter version 2.0.5 at /usr/local/Caskroom/flutter/1.22.6/flutter
    • Framework revision adc687823a (5 days ago), 2021-04-16 09:40:20 -0700
    • Engine revision b09f014e96
    • Dart version 2.12.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/roshanalexander/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/bin/java
    • Java version Java(TM) SE Runtime Environment (build 1.8.0_281-b09)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[✓] IntelliJ IDEA Ultimate Edition (version 2021.1)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 55.1.5
    • Dart plugin version 211.6693.108

[✓] VS Code (version 1.55.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (2 available)
    • iPhone 12 Pro Max (mobile) • 3B2242B1-50C8-439F-B373-315C9E3EF4A1 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 89.0.4389.128

! Doctor found issues in 1 category.

```

**Pod File:**
[Podfile.txt](https://github.com/flutter/flutter/files/6353737/Podfile.txt)

-----------------------------------------------------------------------------------------------------------------------


Next time I observe the issue, I'll post the other files you requested. 

From my previous comment, this is how the error looks like: [Here](https://github.com/flutter/flutter/issues/61063#issuecomment-655560723)

**Comment by jmagman** [comment]
> From my previous comment, this is how the error looks like: Here

Unfortunately these linker issues aren't actionable without full `flutter --verbose` so we can see the linker flags.  So what you had in https://github.com/flutter/flutter/issues/61063#issuecomment-655568808 was good.

> Next time I observe this issue, I'll post the other files you requested.

Thank you!

**Comment by Roshanalexander** [comment]
> Unfortunately these linker issues aren't actionable without full `flutter --verbose` so we can see the linker flags. So what you had in [#61063 (comment)](https://github.com/flutter/flutter/issues/61063#issuecomment-655568808) was good.
> 


@jmagman 
Ha! I was able to reproduce it again.

```
Launching lib/main.dart on iPhone 12 Pro Max in debug mode...
Running Xcode build...
Xcode build done.                                           24.4s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    /Users/roshanalexander/Documents/Personal/Personal Projects/project/ios/Runner/AppDelegate.swift:13:58: warning: conditional cast from 'AppDelegate' to 'UNUserNotificationCenterDelegate' always succeeds
          UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
                                                             ^
    ld: framework not found fluttertoast
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.4.99. (in target 'nanopb' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.4.99. (in target 'Protobuf' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.4.99. (in target 'PromisesObjC' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.4.99. (in target 'GoogleUtilities' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.4.99. (in target 'GoogleDataTransport' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.4.99. (in target 'GoogleDataTransportCCTSupport' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.4.99. (in target 'FirebaseCoreDiagnostics' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.4.99. (in target 'FirebaseCore' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.4.99. (in target 'FirebaseInstallations' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.4.99. (in target 'FirebaseInstanceID' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.4.99. (in target 'FirebaseMessaging' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.4.99. (in target 'FirebaseAnalyticsInterop' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.4.99. (in target 'FirebaseCoreDiagnosticsInterop' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.4.99. (in target 'GoogleAppMeasurement' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.4.99. (in target 'Flutter' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.4.99. (in target 'FirebaseAnalytics' from project 'Pods')
    warning: The iOS Simulator deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 8.0, but the range of supported deployment target versions is 9.0 to 14.4.99. (in target 'Firebase' from project 'Pods')

Could not build the application for the simulator.
Error launching application on iPhone 12 Pro Max.
```

**Here's the flutter run verbose output:**
[verbose_output.txt](https://github.com/flutter/flutter/files/6353933/verbose_output.txt)

**Comment by jmagman** [comment]
@Roshanalexander What are the contents of `.flutter-plugins-dependencies`?  That's what has the mapping of your packages to where they pub caches them.
And what's the output of `ls -la ios/.symlinks/plugins/`?  We're looking for where fluttertoast is mapped to, and if that directory exists.

**Comment by Roshanalexander** [comment]
> What are the contents of `.flutter-plugins-dependencies`? That's what has the mapping of your packages to wher pub caches them.

[flutter-plugins-dependencies.txt](https://github.com/flutter/flutter/files/6354188/flutter-plugins-dependencies.txt)



> And what's the output of `ls -la ios/.symlinks/plugins/`? We're looking for where fluttertoast is mapped to, and if that directory exists.

```
roshanalexander@Roshans-MBP Project % ls -la ios/.symlinks/plugins/
total 0
drwxr-xr-x  18 roshanalexander  staff  576 Apr 21 17:26 .
drwxr-xr-x   3 roshanalexander  staff   96 Apr 21 17:26 ..
lrwxr-xr-x   1 roshanalexander  staff   97 Apr 21 17:26 app_settings -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/app_settings-4.1.0/
lrwxr-xr-x   1 roshanalexander  staff   98 Apr 21 17:26 audioplayers -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.17.4/
lrwxr-xr-x   1 roshanalexander  staff   96 Apr 21 17:26 device_info -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/device_info-2.0.0/
lrwxr-xr-x   1 roshanalexander  staff  104 Apr 21 17:26 firebase_analytics -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_analytics-5.0.16/
lrwxr-xr-x   1 roshanalexander  staff  104 Apr 21 17:26 firebase_messaging -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-6.0.16/
lrwxr-xr-x   1 roshanalexander  staff  105 Apr 21 17:26 flutter_email_sender -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_email_sender-5.0.0/
lrwxr-xr-x   1 roshanalexander  staff  108 Apr 21 17:26 flutter_full_pdf_viewer -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_full_pdf_viewer-1.0.6/
lrwxr-xr-x   1 roshanalexander  staff  104 Apr 21 17:26 flutter_fullpdfview -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_fullpdfview-1.1.5/
lrwxr-xr-x   1 roshanalexander  staff  114 Apr 21 17:26 flutter_local_notifications -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_local_notifications-1.5.0+1/
lrwxr-xr-x   1 roshanalexander  staff   97 Apr 21 17:26 fluttertoast -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/fluttertoast-8.0.6/
lrwxr-xr-x   1 roshanalexander  staff  109 Apr 21 17:26 notification_permissions -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/notification_permissions-0.5.0/
lrwxr-xr-x   1 roshanalexander  staff   97 Apr 21 17:26 package_info -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/package_info-2.0.0/
lrwxr-xr-x   1 roshanalexander  staff   99 Apr 21 17:26 path_provider -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider-1.6.28/
lrwxr-xr-x   1 roshanalexander  staff   90 Apr 21 17:26 share -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/share-2.0.1/
lrwxr-xr-x   1 roshanalexander  staff  104 Apr 21 17:26 shared_preferences -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/shared_preferences-0.5.10/
lrwxr-xr-x   1 roshanalexander  staff   97 Apr 21 17:26 url_launcher -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.5.0/

```

**Comment by Roshanalexander** [comment]
@jmagman 

> @Roshanalexander What are the contents of `.flutter-plugins-dependencies`? That's what has the mapping of your packages to wher pub caches them.

[flutter-plugins-dependencies.txt](https://github.com/flutter/flutter/files/6354204/flutter-plugins-dependencies.txt)

> And what's the output of `ls -la ios/.symlinks/plugins/`? We're looking for where fluttertoast is mapped to, and if that directory exists.

```
roshanalexander@Roshans-MBP Segtho % ls -la ios/.symlinks/plugins/
total 0
drwxr-xr-x  17 roshanalexander  staff  544 Apr 21 18:05 .
drwxr-xr-x   3 roshanalexander  staff   96 Apr 21 18:05 ..
lrwxr-xr-x   1 roshanalexander  staff   97 Apr 21 18:05 app_settings -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/app_settings-4.1.0/
lrwxr-xr-x   1 roshanalexander  staff   98 Apr 21 18:05 audioplayers -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/audioplayers-0.17.4/
lrwxr-xr-x   1 roshanalexander  staff   96 Apr 21 18:05 device_info -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/device_info-2.0.0/
lrwxr-xr-x   1 roshanalexander  staff  104 Apr 21 18:05 firebase_analytics -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_analytics-5.0.16/
lrwxr-xr-x   1 roshanalexander  staff  104 Apr 21 18:05 firebase_messaging -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_messaging-6.0.16/
lrwxr-xr-x   1 roshanalexander  staff  105 Apr 21 18:05 flutter_email_sender -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_email_sender-5.0.0/
lrwxr-xr-x   1 roshanalexander  staff  108 Apr 21 18:05 flutter_full_pdf_viewer -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_full_pdf_viewer-1.0.6/
lrwxr-xr-x   1 roshanalexander  staff  104 Apr 21 18:05 flutter_fullpdfview -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_fullpdfview-1.1.5/
lrwxr-xr-x   1 roshanalexander  staff  114 Apr 21 18:05 flutter_local_notifications -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_local_notifications-1.5.0+1/
lrwxr-xr-x   1 roshanalexander  staff  109 Apr 21 18:05 notification_permissions -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/notification_permissions-0.5.0/
lrwxr-xr-x   1 roshanalexander  staff   97 Apr 21 18:05 package_info -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/package_info-2.0.0/
lrwxr-xr-x   1 roshanalexander  staff   99 Apr 21 18:05 path_provider -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider-1.6.28/
lrwxr-xr-x   1 roshanalexander  staff   90 Apr 21 18:05 share -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/share-2.0.1/
lrwxr-xr-x   1 roshanalexander  staff  104 Apr 21 18:05 shared_preferences -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/shared_preferences-0.5.10/
lrwxr-xr-x   1 roshanalexander  staff   97 Apr 21 18:05 url_launcher -> /usr/local/Caskroom/flutter/1.22.6/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.5.0/

```

**Comment by jmagman** [comment]
@Roshanalexander Interesting, it's not in either.  And just to confirm, `fluttertoast` isn't in your pubspec right now?  https://github.com/flutter/flutter/issues/61063#issuecomment-658486469 has it but your original report says it's been removed.

Can you attach:
1. ios/Podfile.lock
2. ios/Runner.xcodeproj/project.pbxproj 
3. ios/Pods/Target Support Files/Pods-Runner/Pods-Runner.debug.xcconfig

**Comment by Roshanalexander** [comment]
> @Roshanalexander Interesting, it's not in either.  And just to confirm, `fluttertoast` isn't in your pubspec right now?  https://github.com/flutter/flutter/issues/61063#issuecomment-658486469 has it but your original report says it's been removed.
> 

I removed it from my pubspec file to reproduce the issue. If I put it back, then the iOS build works fine. Fails when I remove it.

**Comment by jmagman** [comment]
@Roshanalexander Next time you're reproducing can you attach:

> 1. ios/Podfile.lock
> 2. ios/Runner.xcodeproj/project.pbxproj
> 3. ios/Pods/Target Support Files/Pods-Runner/Pods-Runner.debug.xcconfig

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
